pkg/test/framework/api/components/galley.go

```diff
@@ -35,7 +43,153 @@ type Galley interface {
 	SetMeshConfig(yamlText string) error
 
 	// WaitForSnapshot waits until the given snapshot is observed for the given type URL.
-	WaitForSnapshot(collection string, snapshot ...map[string]interface{}) error
+	WaitForSnapshot(collection string, validator SnapshotValidator) error
```

Comment:
Can't this be a simple function? I was expecting this to be a simple function that is passed by the specific tests with specific validation logic.

Reply:
Ah, I think if you make the argument to the validator the full set, of objects, instead of single object than it should work as I described.

Reply:
done.